src/rest_vol.c

```diff
@@ -2404,6 +2302,186 @@ RV_find_object_by_path(RV_object_t *parent_obj, const char *obj_path, H5I_type_t
     return ret_value;
 } /* end RV_find_object_by_path() */
 
+/*-------------------------------------------------------------------------
+ * Function:    RV_parse_creation_properties_callback
+ *
+ * Purpose:     Helper function to try to parse creation properties
+ *              from given parse tree, into ptr at address GCPL_buf.
+ *
+ *              This is a separate function to allow it to fail
+ *              gracefully when the object we're operating on
+ *              doesn't have a creation properties field.
+ *
+ *              If it succeeds, it will allocate memory at
+ *              *GCPL_buf that must be freed by user.
+ *
+ * Return:      Non-negative on success/Negative on failure
+ *
+ * Programmer:  Matthew Larson
+ *              May, 2023
+ */
+herr_t
+RV_parse_creation_properties_callback(yajl_val parse_tree, char **GCPL_buf)
+{
+    herr_t   ret_value     = SUCCEED;
+    yajl_val key_obj       = NULL;
+    char    *parsed_string = NULL;
+
+    if (!parse_tree)
+        FUNC_GOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "parse tree was NULL");
+
+    if (NULL == (key_obj = yajl_tree_get(parse_tree, object_creation_properties_keys, yajl_t_string)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_BADVALUE, FAIL, "failed to parse creationProperties");
+
+    if (!YAJL_IS_STRING(key_obj))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_BADVALUE, FAIL, "returned creationProperties is not a string");
+
+    if (NULL == (parsed_string = YAJL_GET_STRING(key_obj)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_BADVALUE, FAIL, "creationProperties was NULL");
+
+    if (NULL == (*GCPL_buf = RV_malloc(strlen(parsed_string) + 1)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_CANTALLOC, FAIL, "failed to allocate memory for creationProperties");
+
+    if (NULL == (memcpy(*GCPL_buf, parsed_string, strlen(parsed_string) + 1)))
+        FUNC_GOTO_ERROR(H5E_OBJECT, H5E_SYSERRSTR, FAIL, "failed to copy creationProperties");
+
+done:
+
+    if (ret_value < 0) {
+        free(*GCPL_buf);
```

Comment:
I don't think this is important enough to hold this PR, but note that this function should probably be updated to allocate the GCPL buf and assign it to a local variable that was first initialized to NULL, then make the free call here on the local variable rather than the `GCPL_buf` pointer passed in. Otherwise, a failure in this function before the call to `RV_malloc` will cause a segfault or double free error if the caller didn't initialize `*GCPL_buf` to NULL. Typically, to prevent disturbing the caller's state on failure, I'll wait to assign the local pointer to the returned one until the last statement in the body of the function.